Tests

Elisa-Visentin opened the pull request:
Here some test functions (for the `io` module): real data and MCs are processed from DL0s up to DL3s/IRFs and provided to the `load` functions. Still to be adapted to the 4LSTs updates and to Git workflows (and maybe to the ctapipe 0.19 upgrades); now they can be run 'locally'. Test files now in my workspace on the IT Container; they will then be moved to a location providing, e.g., `wget` downloads with Git secrets.

In the meanwhile...
        Comments? suggestions? errors? 
        Somebody wants to try to run them? (Maybe I need to tell you how to do this...)

See also #137 



**STILL NOT TO BE MERGED** (Maybe after the 4LSTs upgrades)

jsitarek commented on magicctapipe/scripts/lst1_magic/config.yaml:
```diff
@@ -200,7 +200,7 @@ event_classifier:
 create_irf:
     quality_cuts: "disp_diff_mean < 0.22"
     event_type: "software"  # select "software", "software_only_3tel", "magic_only" or "hardware"
-    weight_type_dl2: "intensity"  # select "simple", "variance" or "intensity"
```
I think this change was not intended (the intensity weight is our standard since it provides a bit better performance)

Elisa-Visentin replied:
Hi, This is the config file used by the tests, and I used 'simple' because, this way, I can check if the results are the expected ones (I can evaluate simple averages by hand and check if the values returned by the 'load' functions are the right ones). Obviously, tests only check the algorithms; the results can't be used to get to some scientific result, so the performance is not really important (anyway, with one LST sub-run and one MC run, the spectrum and the IRFs are meaningless). Before merging, we will have to discuss about where to put this config (with data instead of in the repo? or in the repo but with a different name?)

jsitarek replied:
hmmm, something is wrong here. 
The actual config file for tests you also download together with the data (in config subdirectory), so you do not need to include it in the repository. Since config files are small we could do it if needed, but the way how you did it, it overwrites the default config, and this is bad idea, both because this config file is used by the users, and also the default config will be also modified in the future following improvements of the analysis, and then you would need to redo all the tests with the new config

Elisa-Visentin replied:
Yes I can remove this from the PR (not to overwrite it)

aleberti replied:
Now the configuration files are in the repository, but they are (should be) used as they are, without overwriting any exiting configuration file.

jsitarek commented on magicctapipe/conftest.py:
```diff
@@ -0,0 +1,660 @@
+    """
+    Download of test files through rsync
+    """
+    pwd = os.environ[env_prefix + "PASSWORD"]
```
you can exchange the ssh keys with the IT cluster, which allows you to login without a password, and thus will avoid putting the password in an environment variable (that is not very secure)

Elisa-Visentin replied:
This is a temporary solution, before merging it we will have to find a location for the test files (now they are in my workspace, not a good idea because we have to use subprocess.run; if we put them with the MAGIC ones on the same server used my ctapipe-io-magic, we can download them with a dedicated ctapipe function that automatically manages some keys that can be stored as Git Secrets). SSH keys are a big issue because there is no 'test' global user (i.e.: password and username shared by all the Collaboration) and everyone has his keys and his tunnels (and an ssh config file). Public ssh keys can be used to skip passwords, but I don't know if they can be used (again, security issues). Maybe private keys? 
While waiting for data to be stored somewhere, this allows people to test the tests, but for sure we will have to change this. Unfortunately, to access the IT Container I need more than one ProxyJump (and some public and private keys), so that Paramiko can't be used. This is why we must not merge it now :)

jsitarek replied:
the keys would work find for individual users, but indeed for implementation in the repository itself for automatic tests this can be a problem
if we can use the same solution as ctapipe_io_magic, this would be preferred.

Nevertheless, the whole pack of files to download is 2 GB, and this is quite some size, plus it makes the processing time long. The largest files are the gamma MCs (2 x 400 MB), and DL1 from LST (1 subrun of 700 MB) It would be good to use dedicated gamma MCs (either ask Georgios to provide two files with e.g. a factor of 30 less events per file,  or maybe  try to cut the file ourselfs. 
Similar with the data file - to find a much smaller LST subrun we can always take the last subrun of a given run.

Elisa-Visentin replied:
As for MCs, for sure we can ask for smaller datasets (I can try to ask Georgios, maybe he won't curse me) while waiting for the permissions needed to upload them to the MAGIC server.
As for the LST sub-run, I will have to do some tests because there must be some coincident events ('a lot' because they have to survive quality cuts up to DL3), and the wobble shifts are not synchronous... 
I don't know if we can cut the files (w/o running into exceptions or errors), but I can take a look at this, too.

aleberti replied:
no need for the `scp`/`rsync` function anymore, we use now the ctapipe `download_file_cached` function, after the data have been moved to the MAGIC webserver (btw, also this is temporary, possibly we will put them at PIC).

jsitarek commented on magicctapipe/conftest.py:
```diff
@@ -0,0 +1,660 @@
+            print("files cannot be removed")
+            exit()
+        usr = os.environ[env_prefix + "USER"]
+        childP = pexpect.spawn(cmd, timeout=5000)
```
at least in my machine rsync does not create the needed parent directory, so in order to make the files being downloaded I needed to add:
```subprocess.run(["mkdir", "-p", str(path)])```

Elisa-Visentin replied:
This is another issue due to the SSH connection (IT Container). Test data should be downloaded to a temporary directory created on the fly while running the tests and deleted at the end of the session (as the ones used to store data). Now, the IT Container is really busy and often runs into crashes, so, to avoid to keep on downloading the same data (the administrators won't be happy) I coded more 'persistent' downloads. 
But I will fix the directory creation (I didn't noticed this, thank you really much :)  )

jsitarek commented on magicctapipe/conftest.py:
```diff
@@ -0,0 +1,662 @@
+@pytest.fixture(scope="session")
+def base_path():
+    # TO BE CHANGED
+    return Path("/home/elisa/.cache/magicctapipe")
```
return Path(str(os.environ["HOME"])+"/.cache/magicctapipe")

jsitarek commented on magicctapipe/io/tests/test_io.py:
```diff
@@ -0,0 +1,418 @@
+        assert point[0] <= 90
+
+
+def test_load_mc_dl2_data_file_cut(p_dl2, gamma_dl2):
```
still there with the new files...

aleberti replied:
no need for the `scp`/`rsync` function anymore, we use now the ctapipe `download_file_cached` function, after the data have been moved to the MAGIC webserver (btw, also this is temporary, possibly we will put them at PIC).

Elisa-Visentin commented:
Still failed tests?

Added MAGIC-only tests (separated from the joint analysis ones)
MC: << number of showers.

Added input variable to magic_calib_to_dl1.py: maximum number of showers processed -> still 13000 showers to be processed in case of joint analysis: maybe adding another variable to 'shrink' MAGIC files up to approx. 500/1000 showers?

Any other suggestion, error...?

jsitarek commented:
now it works like a charm for me, no broken tests any more

aleberti commented:
is it normal that download takes ages? what is the total size of the files?

Elisa-Visentin commented:
Great. Calibrated MAGIC data still require a few minutes to be processed,
but no coincident event can be found if we process less than 13000 showers
(13500= number of events in the subrun), so, 2 options:
1) Wait
2) Modify a bit  the MCP script that processed them to allow it to process
only a range of events (e.g., from 12000 to 13000).

As for the other tests, they have to process data from DL0 up to DL1-stereo
(or DL2) and to apply, e.g., the get_dl2_mean, so a lot of algorithms have
to be applied... (and we cannot use smaller MC p files because we discard a
lot of p events during the analysis and we need them up to DL2)

Il giorno mar 11 lug 2023 alle ore 10:24 Julian Sitarek <
***@***.***> ha scritto:

> Duration of individual tests (after all the files are downloaded):
>
> 306.54s setup    magicctapipe/io/tests/test_io.py::test_load_magic_dl1_data_files
> 151.37s setup    magicctapipe/io/tests/test_io.py::test_get_stereo_events_mc
> 47.05s setup    magicctapipe/io/tests/test_io.py::test_load_mc_dl2_data_file
> 27.61s setup    magicctapipe/io/tests/test_io.py::test_get_stereo_events_data
> 13.91s setup    magicctapipe/io/tests/test_io.py::test_index
> 8.61s setup    magicctapipe/io/tests/test_io.py::test_load_dl2_data_file
> 6.58s setup    magicctapipe/io/tests/test_io.py::test_load_irf_files
> 1.21s call     magicctapipe/io/tests/test_io.py::test_get_dl2_mean_avg
> 0.58s call     magicctapipe/io/tests/test_io.py::test_get_stereo_events_mc
> 0.57s call     magicctapipe/io/tests/test_io.py::test_get_stereo_events_mc_cut
> 0.36s call     magicctapipe/io/tests/test_io.py::test_load_mc_dl2_data_file
> 0.35s call     magicctapipe/io/tests/test_io.py::test_load_mc_dl2_data_file_cut
> 0.34s call     magicctapipe/io/tests/test_io.py::test_load_mc_dl2_data_file_opt
> 0.22s call     magicctapipe/io/tests/test_io.py::test_load_mc_dl2_data_file_exc
> 0.22s call     magicctapipe/io/tests/test_io.py::test_load_dl2_data_file
> 0.21s call     magicctapipe/io/tests/test_io.py::test_load_dl2_data_file_opt
> 0.21s call     magicctapipe/io/tests/test_io.py::test_load_dl2_data_file_cut
> 0.16s call     magicctapipe/io/tests/test_io.py::test_load_train_data_files_off
> 0.15s call     magicctapipe/io/tests/test_io.py::test_load_train_data_files_g
> 0.13s call     magicctapipe/io/tests/test_io.py::test_get_dl2_mean_exc
> 0.13s call     magicctapipe/io/tests/test_io.py::test_get_dl2_mean_real
> 0.12s call     magicctapipe/io/tests/test_io.py::test_load_train_data_files_p
> 0.12s call     magicctapipe/io/tests/test_io.py::test_get_dl2_mean_mc
> 0.11s call     magicctapipe/io/tests/test_io.py::test_load_lst_dl1_data_file
> 0.11s call     magicctapipe/io/tests/test_io.py::test_load_dl2_data_file_exc
> 0.10s call     magicctapipe/io/tests/test_io.py::test_load_magic_dl1_data_files
> 0.09s call     magicctapipe/io/tests/test_io.py::test_get_stereo_events_data
> 0.09s call     magicctapipe/io/tests/test_io.py::test_get_stereo_events_data_cut
> 0.04s teardown magicctapipe/io/tests/test_io.py::test_index
> 0.03s call     magicctapipe/io/tests/test_io.py::test_load_irf_files
> 0.02s setup    magicctapipe/io/tests/test_io.py::test_get_dl2_mean_avg
> 0.01s call     magicctapipe/io/tests/test_io.py::test_save_pandas_data_in_table
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cta-observatory/magic-cta-pipe/pull/141#issuecomment-1630377432>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A43O4NDXR7OQI2EQVGXDDQTXPUEVRANCNFSM6AAAAAAXY732XA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
------------------------


Indirizzo istituzionale di posta elettronica 
degli studenti e dei laureati dell'Università di TorinoOfficial University 
of Turin email address for students and graduates 

Elisa-Visentin commented:
The total size is < 500 MB, but the download is not really efficient (and
it depends on the IT Container status, on the ssh connection...

Il giorno mar 11 lug 2023 alle ore 11:24 Elisa Visentin <
***@***.***> ha scritto:

> Great. Calibrated MAGIC data still require a few minutes to be processed,
> but no coincident event can be found if we process less than 13000 showers
> (13500= number of events in the subrun), so, 2 options:
> 1) Wait
> 2) Modify a bit  the MCP script that processed them to allow it to process
> only a range of events (e.g., from 12000 to 13000).
>
> As for the other tests, they have to process data from DL0 up to
> DL1-stereo (or DL2) and to apply, e.g., the get_dl2_mean, so a lot of
> algorithms have to be applied... (and we cannot use smaller MC p files
> because we discard a lot of p events during the analysis and we need them
> up to DL2)
>
> Il giorno mar 11 lug 2023 alle ore 10:24 Julian Sitarek <
> ***@***.***> ha scritto:
>
>> Duration of individual tests (after all the files are downloaded):
>>
>> 306.54s setup    magicctapipe/io/tests/test_io.py::test_load_magic_dl1_data_files
>> 151.37s setup    magicctapipe/io/tests/test_io.py::test_get_stereo_events_mc
>> 47.05s setup    magicctapipe/io/tests/test_io.py::test_load_mc_dl2_data_file
>> 27.61s setup    magicctapipe/io/tests/test_io.py::test_get_stereo_events_data
>> 13.91s setup    magicctapipe/io/tests/test_io.py::test_index
>> 8.61s setup    magicctapipe/io/tests/test_io.py::test_load_dl2_data_file
>> 6.58s setup    magicctapipe/io/tests/test_io.py::test_load_irf_files
>> 1.21s call     magicctapipe/io/tests/test_io.py::test_get_dl2_mean_avg
>> 0.58s call     magicctapipe/io/tests/test_io.py::test_get_stereo_events_mc
>> 0.57s call     magicctapipe/io/tests/test_io.py::test_get_stereo_events_mc_cut
>> 0.36s call     magicctapipe/io/tests/test_io.py::test_load_mc_dl2_data_file
>> 0.35s call     magicctapipe/io/tests/test_io.py::test_load_mc_dl2_data_file_cut
>> 0.34s call     magicctapipe/io/tests/test_io.py::test_load_mc_dl2_data_file_opt
>> 0.22s call     magicctapipe/io/tests/test_io.py::test_load_mc_dl2_data_file_exc
>> 0.22s call     magicctapipe/io/tests/test_io.py::test_load_dl2_data_file
>> 0.21s call     magicctapipe/io/tests/test_io.py::test_load_dl2_data_file_opt
>> 0.21s call     magicctapipe/io/tests/test_io.py::test_load_dl2_data_file_cut
>> 0.16s call     magicctapipe/io/tests/test_io.py::test_load_train_data_files_off
>> 0.15s call     magicctapipe/io/tests/test_io.py::test_load_train_data_files_g
>> 0.13s call     magicctapipe/io/tests/test_io.py::test_get_dl2_mean_exc
>> 0.13s call     magicctapipe/io/tests/test_io.py::test_get_dl2_mean_real
>> 0.12s call     magicctapipe/io/tests/test_io.py::test_load_train_data_files_p
>> 0.12s call     magicctapipe/io/tests/test_io.py::test_get_dl2_mean_mc
>> 0.11s call     magicctapipe/io/tests/test_io.py::test_load_lst_dl1_data_file
>> 0.11s call     magicctapipe/io/tests/test_io.py::test_load_dl2_data_file_exc
>> 0.10s call     magicctapipe/io/tests/test_io.py::test_load_magic_dl1_data_files
>> 0.09s call     magicctapipe/io/tests/test_io.py::test_get_stereo_events_data
>> 0.09s call     magicctapipe/io/tests/test_io.py::test_get_stereo_events_data_cut
>> 0.04s teardown magicctapipe/io/tests/test_io.py::test_index
>> 0.03s call     magicctapipe/io/tests/test_io.py::test_load_irf_files
>> 0.02s setup    magicctapipe/io/tests/test_io.py::test_get_dl2_mean_avg
>> 0.01s call     magicctapipe/io/tests/test_io.py::test_save_pandas_data_in_table
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/cta-observatory/magic-cta-pipe/pull/141#issuecomment-1630377432>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A43O4NDXR7OQI2EQVGXDDQTXPUEVRANCNFSM6AAAAAAXY732XA>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

-- 
------------------------


Indirizzo istituzionale di posta elettronica 
degli studenti e dei laureati dell'Università di TorinoOfficial University 
of Turin email address for students and graduates 

aleberti commented:
Just tried now, and everything is fine except for `test_load_train_data_files_off` where it fails with:

```python
gamma_stereo = (PosixPath('/tmp/pytest-of-aberti/pytest-2/DL1_gamma_train0'), PosixPath('/tmp/pytest-of-aberti/pytest-2/DL1_gamma_test0'))

    def test_load_train_data_files_off(gamma_stereo):
        """
        Check off-axis cut
        """
        events = load_train_data_files(
            str(gamma_stereo[0]), offaxis_min="0.2 deg", offaxis_max="0.5 deg"
        )
>       data = events["LST1_M2"]
E       KeyError: 'LST1_M2'

magicctapipe/io/tests/test_io.py:111: KeyError
------------------------------------------------------------ Captured log call -------------------------------------------------------------
INFO     magicctapipe.io.io:io.py:493 
The following DL1-stereo data files are found:
INFO     magicctapipe.io.io:io.py:498 /tmp/pytest-of-aberti/pytest-2/DL1_gamma_train0/dl1_stereo_gamma_zd_16.087deg_az_108.0deg_LST-1_MAGIC_run201.h5
INFO     magicctapipe.io.io:io.py:498 /tmp/pytest-of-aberti/pytest-2/DL1_gamma_train0/dl1_stereo_gamma_zd_16.087deg_az_108.0deg_LST-1_MAGIC_run301.h5
INFO     magicctapipe.io.io:io.py:133 
In total 7 stereo events are found:
INFO     magicctapipe.io.io:io.py:170     M1_M2 (type 0): 0 events (0.0%)
    LST1_M1 (type 1): 2 events (28.6%)
    LST1_M2 (type 2): 0 events (0.0%)
    LST1_M1_M2 (type 3): 5 events (71.4%)

```

By the way, I had to change the IP in `conftest.py` to `cp02` to make the download work.

Elisa-Visentin commented:
Actually, no LST1_M2 events are found. I can change it to LST1_M1_M2 and
test it again (my laptop did not run into this error... 🤔)

Il giorno mer 12 lug 2023 alle ore 11:05 Alessio Berti <
***@***.***> ha scritto:

> Just tried now, and everything is fine except for
> test_load_train_data_files_off where it fails with:
>
> gamma_stereo = (PosixPath('/tmp/pytest-of-aberti/pytest-2/DL1_gamma_train0'), PosixPath('/tmp/pytest-of-aberti/pytest-2/DL1_gamma_test0'))
>
>     def test_load_train_data_files_off(gamma_stereo):
>         """        Check off-axis cut        """
>         events = load_train_data_files(
>             str(gamma_stereo[0]), offaxis_min="0.2 deg", offaxis_max="0.5 deg"
>         )>       data = events["LST1_M2"]E       KeyError: 'LST1_M2'
> magicctapipe/io/tests/test_io.py:111: KeyError------------------------------------------------------------ Captured log call -------------------------------------------------------------INFO     magicctapipe.io.io:io.py:493 The following DL1-stereo data files are found:INFO     magicctapipe.io.io:io.py:498 /tmp/pytest-of-aberti/pytest-2/DL1_gamma_train0/dl1_stereo_gamma_zd_16.087deg_az_108.0deg_LST-1_MAGIC_run201.h5INFO     magicctapipe.io.io:io.py:498 /tmp/pytest-of-aberti/pytest-2/DL1_gamma_train0/dl1_stereo_gamma_zd_16.087deg_az_108.0deg_LST-1_MAGIC_run301.h5INFO     magicctapipe.io.io:io.py:133 In total 7 stereo events are found:INFO     magicctapipe.io.io:io.py:170     M1_M2 (type 0): 0 events (0.0%)
>     LST1_M1 (type 1): 2 events (28.6%)
>     LST1_M2 (type 2): 0 events (0.0%)
>     LST1_M1_M2 (type 3): 5 events (71.4%)
>
> By the way, I had to change the IP in conftest.py to cp02 to make the
> download work.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cta-observatory/magic-cta-pipe/pull/141#issuecomment-1632131677>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A43O4NALJQ2QWSZVLHXGTZTXPZSHJANCNFSM6AAAAAAXY732XA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
------------------------


Indirizzo istituzionale di posta elettronica 
degli studenti e dei laureati dell'Università di TorinoOfficial University 
of Turin email address for students and graduates 

Elisa-Visentin commented:
As for the download, this will always depend on our shh config files, but I
can add a variable to be used as the IP and to be set by the user (as long
as we download data from CP02). I will work on this ASAP

Il giorno mer 12 lug 2023 alle ore 11:30 Elisa Visentin <
***@***.***> ha scritto:

> Actually, no LST1_M2 events are found. I can change it to LST1_M1_M2 and
> test it again (my laptop did not run into this error... 🤔)
>
> Il giorno mer 12 lug 2023 alle ore 11:05 Alessio Berti <
> ***@***.***> ha scritto:
>
>> Just tried now, and everything is fine except for
>> test_load_train_data_files_off where it fails with:
>>
>> gamma_stereo = (PosixPath('/tmp/pytest-of-aberti/pytest-2/DL1_gamma_train0'), PosixPath('/tmp/pytest-of-aberti/pytest-2/DL1_gamma_test0'))
>>
>>     def test_load_train_data_files_off(gamma_stereo):
>>         """        Check off-axis cut        """
>>         events = load_train_data_files(
>>             str(gamma_stereo[0]), offaxis_min="0.2 deg", offaxis_max="0.5 deg"
>>         )>       data = events["LST1_M2"]E       KeyError: 'LST1_M2'
>> magicctapipe/io/tests/test_io.py:111: KeyError------------------------------------------------------------ Captured log call -------------------------------------------------------------INFO     magicctapipe.io.io:io.py:493 The following DL1-stereo data files are found:INFO     magicctapipe.io.io:io.py:498 /tmp/pytest-of-aberti/pytest-2/DL1_gamma_train0/dl1_stereo_gamma_zd_16.087deg_az_108.0deg_LST-1_MAGIC_run201.h5INFO     magicctapipe.io.io:io.py:498 /tmp/pytest-of-aberti/pytest-2/DL1_gamma_train0/dl1_stereo_gamma_zd_16.087deg_az_108.0deg_LST-1_MAGIC_run301.h5INFO     magicctapipe.io.io:io.py:133 In total 7 stereo events are found:INFO     magicctapipe.io.io:io.py:170     M1_M2 (type 0): 0 events (0.0%)
>>     LST1_M1 (type 1): 2 events (28.6%)
>>     LST1_M2 (type 2): 0 events (0.0%)
>>     LST1_M1_M2 (type 3): 5 events (71.4%)
>>
>> By the way, I had to change the IP in conftest.py to cp02 to make the
>> download work.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/cta-observatory/magic-cta-pipe/pull/141#issuecomment-1632131677>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A43O4NALJQ2QWSZVLHXGTZTXPZSHJANCNFSM6AAAAAAXY732XA>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

-- 
------------------------


Indirizzo istituzionale di posta elettronica 
degli studenti e dei laureati dell'Università di TorinoOfficial University 
of Turin email address for students and graduates 

jsitarek commented:
it is funny that it worked for me, but failed for Alessio, unless the file was changed in the meantime.

concerning the connection, I think now we have an agreement to put the files in DOrtmund machine (like it was done for ctapipe_io_magic), so we can switch to it (and merge the PR :-) )

Elisa-Visentin commented:
It worked for me too. Maybe somebody else could try them?

BTW, we MUST move the files within the end of August (my affiliations will
expire and I don't know if my workspace will be deleted....)

Il giorno mer 12 lug 2023 alle ore 11:51 Julian Sitarek <
***@***.***> ha scritto:

> it is funny that it worked for me, but failed for Alessio, unless the file
> was changed in the meantime.
>
> concerning the connection, I think now we have an agreement to put the
> files in DOrtmund machine (like it was done for ctapipe_io_magic), so we
> can switch to it (and merge the PR :-) )
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cta-observatory/magic-cta-pipe/pull/141#issuecomment-1632200107>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A43O4NFMDTFB5DUYD7B5ETTXPZXRZANCNFSM6AAAAAAXY732XA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
------------------------


Indirizzo istituzionale di posta elettronica 
degli studenti e dei laureati dell'Università di TorinoOfficial University 
of Turin email address for students and graduates 

jsitarek commented:
If the files are copied, and we switch on CI tests (like in here https://github.com/cta-observatory/ctapipe_io_magic/blob/master/.github/workflows/ci.yml) , than the github will do the actual test, and this one is the most important.

Elisa-Visentin commented:
We can wait and, if needed, fix them to adapt them to Git workflows. Let me
know if you want me to fix them (or to fix something else) before the
merging

Il giorno mer 12 lug 2023 alle ore 12:01 Julian Sitarek <
***@***.***> ha scritto:

> If the files are copied, and we switch on CI tests (like in here
> https://github.com/cta-observatory/ctapipe_io_magic/blob/master/.github/workflows/ci.yml)
> , than the github will do the actual test, and this one is the most
> important.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cta-observatory/magic-cta-pipe/pull/141#issuecomment-1632214455>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A43O4NCY63LWX4WFGJB7MTDXPZYXVANCNFSM6AAAAAAXY732XA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
------------------------


Indirizzo istituzionale di posta elettronica 
degli studenti e dei laureati dell'Università di TorinoOfficial University 
of Turin email address for students and graduates 

jsitarek commented:
I run again the tests, because I realized that the MAGIC-only I haven't run yet. 61 passes (with 8 warnings, but no fails)

the changes to copy the files from Dortmund instead of the IT cluster seems easy, I am just not sure where in github to put the secret credentials, maybe we need @aleberti for this. I propose that we do it now, and check if it works with CI

Elisa-Visentin commented:
The repo administrator can access all of the settings and, among them,
there is something like 'action secrets'. We only need to adapt the paths
and the download function

Il giorno mer 12 lug 2023 alle ore 12:26 Julian Sitarek <
***@***.***> ha scritto:

> I run again the tests, because I realized that the MAGIC-only I haven't
> run yet. 61 passes (with 8 warnings, but no fails)
>
> the changes to copy the files from Dortmund instead of the IT cluster
> seems easy, I am just not sure where in github to put the secret
> credentials, maybe we need @aleberti <https://github.com/aleberti> for
> this. I propose that we do it now, and check if it works with CI
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cta-observatory/magic-cta-pipe/pull/141#issuecomment-1632247844>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A43O4NC6CA3AJXMHFGZE6FTXPZ3UFANCNFSM6AAAAAAXY732XA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
------------------------


Indirizzo istituzionale di posta elettronica 
degli studenti e dei laureati dell'Università di TorinoOfficial University 
of Turin email address for students and graduates 

jsitarek commented:
The secrets are related to the repository, and as far as I understand
everybody is fine with putting the files in Dortmund (right?), so let's
copy them and implement download from them, such that we can enable CI tests

śr., 12 lip 2023, 13:50 użytkownik Alessio Berti ***@***.***>
napisał:

> I can add the secrets in the repository, but I did not understand if we
> want to have first the files moved to Dortmund.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cta-observatory/magic-cta-pipe/pull/141#issuecomment-1632386976>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH36DYJGCUFY23LDCQIYN5DXP2FOXANCNFSM6AAAAAAXY732XA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

aleberti commented:
The LST files of course, the MAGIC files I will put them in the MAGIC webserver in LP, that's what was blessed in MAGIC EB. Some change needed, and two credentials to add, but not an issue. For Dortmund, I guess we need to ask Max to put the files.

Elisa-Visentin commented:
The download functions take as an argument the complete path
(Dortmund/MAGIC+subfolders), so we only need to define two base paths
instead of one and to change the path fixtures so that they return the
right path (LSM from Dortmund, MAGIC from its server). As for the keys, I'm
not an expert but it should be possible to save in Git two different keys
(and the download function must be provided path+related keys, so every
time we download a file we only have to do something like '{download
function}(Dortmund path, Dortmund keys, local path)' or '{download
function}(MAGIC path, MAGIC keys, local path)'

Il giorno mer 12 lug 2023 alle ore 14:03 Alessio Berti <
***@***.***> ha scritto:

> The LST files of course, the MAGIC files I will put them in the MAGIC
> webserver in LP, that's what was blessed in MAGIC EB. Some change needed,
> and two credentials to add, but not an issue. For Dortmund, I guess we need
> to ask Max to put the files.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cta-observatory/magic-cta-pipe/pull/141#issuecomment-1632404098>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A43O4NFYRNZ3U4FALVKS74TXP2G73ANCNFSM6AAAAAAXY732XA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
------------------------


Indirizzo istituzionale di posta elettronica 
degli studenti e dei laureati dell'Università di TorinoOfficial University 
of Turin email address for students and graduates 

aleberti commented:
I uploaded the MAGIC data files at:
https://www.magic.iac.es/mcp-testdata/test_data/real/calibrated/20201216_M1_05093711.014_Y_CrabNebula-W0.40+035.root
https://www.magic.iac.es/mcp-testdata/test_data/real/calibrated/20201216_M2_05093711.014_Y_CrabNebula-W0.40+035.root

I also added the repository secrets `MAGIC_TEST_DATA_USER`, `MAGIC_TEST_DATA_PASSWORD`, `LST_TEST_DATA_USER` and `LST_TEST_DATA_PASSWORD` to be used in the CI.

jsitarek commented:
Thank you @aleberti !
Concerning the lack of M1+M2 events this might be plain statistics, because those events are more rare than the types with LST. But it would be good to have at least one to avoid a possible problem in the future. To do it without modifying the files we could change e.g. the intensity cut used in the test  (either to lower value to have some events that did not survive earlier, or it might also help to set it to higher value if some even loses LST-1 image then. However we should then check that there is no issue on further tests.

aleberti commented:
> Thank you @aleberti ! Concerning the lack of M1+M2 events this might be plain statistics, because those events are more rare than the types with LST. But it would be good to have at least one to avoid a possible problem in the future. To do it without modifying the files we could change e.g. the intensity cut used in the test (either to lower value to have some events that did not survive earlier, or it might also help to set it to higher value if some even loses LST-1 image then. However we should then check that there is no issue on further tests.

ok, now tests are passing, but somehow I am a bit puzzled. The test that is failing for me, never failed in the CI tests here. What failed was the one loading IRF files, where a key called `file_names` is found, but I cannot reproduce it offline. I added it in the list of keys to be checked and of course the tests now are fine.

About the test failing for me, indeed changing the intensity cut can work. In order to make this easier, I will include the configuration files used in the tests directly in the repository, and not download them from the web server. It will make easier to do changes if in the future we change something in configuration files.

aleberti commented:
Ok, implemented to have the configuration files in the repository. I tried a few intensity cuts (even simply >0), but nothing, I always get the same failing test offline. I wonder how it is possible that here it was not failing. Apart from that, everything else is ok. @jsitarek , @Elisa-Visentin , if you are ok with this, I would say it can be merged.

Elisa-Visentin commented:
As for the IRF test failing... I implemented the test functions by using
the SW school pipeline, but a few weeks ago a PR about IRF implementation
was merged. I changed the config file, but I missed the added key
(sorry!!!) so maybe the file_names key must be added in the test (I will
check it now). According to the version you are using, the test will pass
or fail. Before merging, I can check this (any other recent major change?)

Il mar 22 ago 2023, 13:49 Alessio Berti ***@***.***> ha
scritto:

> Ok, implemented to have the configuration files in the repository. I tried
> a few intensity cuts (even simply >0), but nothing, I always get the same
> failing test offline. I wonder how it is possible that here it was not
> failing. Apart from that, everything else is ok. @jsitarek
> <https://github.com/jsitarek> , @Elisa-Visentin
> <https://github.com/Elisa-Visentin> , if you are ok with this, I would
> say it can be merged.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cta-observatory/magic-cta-pipe/pull/141#issuecomment-1688033559>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A43O4NE5HSZVYPMDA6YPVH3XWSMDDANCNFSM6AAAAAAXY732XA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 
------------------------


Indirizzo istituzionale di posta elettronica 
degli studenti e dei laureati dell'Università di TorinoOfficial University 
of Turin email address for students and graduates 

Elisa-Visentin commented:
As for the merging... the coincidence algorithm has been re-implemented since the creation of this branch, so some merge conflicts will have to be solved (and maybe the configuration files read by the tests will have to be changed a bit: 2 lines were added), but the tests should work. BTW, while implementing these tests I did not changed the MCP coincidence script, I only added some files and modified a bit the maigc_calib_to_dl1, not modified in the master since February...).
So, could be merged but config files to be slightly modified (I will do this now)

aleberti commented:
Easy to see if tests work in master: merge it in this branch. And indeed they work.

aleberti commented:
@jsitarek if you also are ok with the PR now, feel free to merge it!

aleberti commented:
Tried to update the CI file with some updated stuff, but better to do it in another PR. So I reverted the changes. I will not touch this PR anymore :)
